how-to/workspace-platform-starter/client/src/framework/utils-position.ts

```diff
@@ -0,0 +1,89 @@
+import type OpenFin from "@openfin/core";
+import { isEmpty } from "./utils";
+
+/**
+ * Provides x and y co-ordinates to position a window of a given size in relation to another window/view.
+ * @param clientIdentity The identity of the view/window these x/y co-ordinates should be in relation to.
+ * @param dimensions The dimensions of the window that will be placed in the center of the screen.
+ * @param dimensions.height The height of the window that is going to be placed
+ * @param dimensions.width The width of the window that is going to be placed.
+ * @returns The x, y co-ordinates to position the window
+ */
+export async function getCenterPosition(
+	clientIdentity: OpenFin.Identity,
+	dimensions: { height: number; width: number }
+): Promise<{ x: number; y: number } | undefined> {
+	const monitorInfo = await getMonitorInfo(clientIdentity);
+	const height = monitorInfo.availableRect.bottom - monitorInfo.availableRect.top;
+	const width = monitorInfo.availableRect.right - monitorInfo.availableRect.left;
+
+	// eslint-disable-next-line no-mixed-operators
+	const x = (monitorInfo.availableRect.left + width) / 2 - dimensions.width / 2;
+	// eslint-disable-next-line no-mixed-operators
+	const y = (monitorInfo.availableRect.top + height) / 2 - dimensions.height / 2;
+
+	return { x, y };
+}
+
+/**
+ * Returns the monitor details for the monitor a view/window is placed on.
+ * @param clientIdentity The identity of the view/window to check against.
+ * @returns The monitor the view/window lives on or undefined if no match was found.
+ */
+export async function getMonitorInfo(clientIdentity: OpenFin.Identity): Promise<OpenFin.MonitorDetails> {
+	let bounds: OpenFin.Bounds | undefined;
+	let currentWindow: OpenFin.Window | undefined;
+
+	try {
+		const targetView = fin.View.wrapSync(clientIdentity);
+		currentWindow = await targetView.getCurrentWindow();
+	} catch {
+		// we were not passed a view.
+	}
+
+	if (isEmpty(bounds)) {
+		try {
+			const targetWindow = currentWindow ?? fin.Window.wrapSync(clientIdentity);
+			bounds = await targetWindow.getBounds();
+		} catch {
+			// it wasn't a window
+		}
+	}
+	if (isEmpty(bounds)) {
+		try {
+			bounds = await fin.me.getBounds();
+		} catch {
+			// unable to get own bounds
+		}
+	}
+
+	const monitorInfo = await fin.System.getMonitorInfo();
```

Comment:
We should probably have some util methods for finding positions in monitors, you will see some very similar methods at the bottom of menu.ts

Reply:
Yeah we should see if we can move that into the util-position file?